src/model/ChillerElectricASHRAE205.cpp

```diff
@@ -0,0 +1,1179 @@
+        return false;
+      }
+
+      auto t_plantLoop = node.plantLoop();
+
+      // Only accept adding to a node that is on a demand side of a plant loop
+      // Since tertiary here = heat recovery loop (heating)
+      if (t_plantLoop) {
+        if (t_plantLoop->demandComponent(node.handle())) {
+          // Call base class method which accepts both supply and demand
+          return WaterToWaterComponent_Impl::addToTertiaryNode(node);
+        } else {
+          LOG(Info,
+              "Tertiary Loop (Heat Recovery Loop) connections can only be placed on the Demand side (of a Heating Loop), for " << briefDescription());
+        }
+      }
+      return false;
+    }
+
+    std::vector<HVACComponent> ChillerElectricASHRAE205_Impl::edges(const boost::optional<HVACComponent>& prev) {
+      // This handles supply, demand, and tertiary connections
+      std::vector<HVACComponent> edges = WaterToWaterComponent_Impl::edges(prev);
+
+      auto pushOilCoolerOutletModelObject = [&]() {
+        if (auto edgeModelObject = oilCoolerOutletModelObject()) {
+          auto edgeHVACComponent = edgeModelObject->optionalCast<HVACComponent>();
+          OS_ASSERT(edgeHVACComponent);
+          edges.push_back(edgeHVACComponent.get());
+        }
+      };
+
+      auto pushAuxiliaryOutletModelObject = [&]() {
+        if (auto edgeModelObject = auxiliaryOutletModelObject()) {
+          auto edgeHVACComponent = edgeModelObject->optionalCast<HVACComponent>();
+          OS_ASSERT(edgeHVACComponent);
+          edges.push_back(edgeHVACComponent.get());
+        }
+      };
+
+      if (prev) {
+        if (auto inletModelObject = oilCoolerInletModelObject()) {
+          if (prev.get() == inletModelObject.get()) {
+            pushOilCoolerOutletModelObject();
+            return edges;
+          }
+        }
+        if (auto inletModelObject = auxiliaryInletModelObject()) {
+          if (prev.get() == inletModelObject.get()) {
+            pushAuxiliaryOutletModelObject();
+            return edges;
+          }
+        }
+      } else {
+        pushOilCoolerOutletModelObject();
+        pushAuxiliaryOutletModelObject();
+        return edges;
+      }
+
+      return edges;
+    }
+
+    boost::optional<PlantLoop> ChillerElectricASHRAE205_Impl::plantLoop() const {
+      if (boost::optional<ModelObject> mo_ = supplyOutletModelObject()) {
+        if (boost::optional<Node> n_ = mo_->optionalCast<Node>()) {
+          return n_->plantLoop();
+        }
+      }
+      return boost::none;
+    }
+
+    boost::optional<PlantLoop> ChillerElectricASHRAE205_Impl::secondaryPlantLoop() const {
+      if (boost::optional<ModelObject> mo_ = demandOutletModelObject()) {
+        if (boost::optional<Node> n_ = mo_->optionalCast<Node>()) {
+          return n_->plantLoop();
+        }
+      }
+      return boost::none;
+    }
```

Comment:
I need to override this one, because by default WaterToWaterComponent will just find a PlantLoop that has it on demand and is not the tertiary... But this object can be on the demand side of **FOUR** loops